### PR TITLE
feat: refdiff adapt new commits_diffs table

### PR DESCRIPTION
**Pull request by abeizn**
### ⚠️ Pre Checklist

> Please complete _ALL_ items in this checklist, and remove before submitting

- [x] I have read through the [Contributing Documentation](https://devlake.apache.org/community/).
- [x] I have added relevant tests.
- [x] I have added relevant documentation.
- [ ] I will add labels to the PR, such as `pr-type/bug-fix`, `pr-type/feature-development`, etc.

# Summary

refdiff adapt new commits_diffs table
1. ref_commit_diff_calculator upgrade
2. fef_issue_diff_calculator upgrade
3. other adapt commits_diffs table

### Does this close any open issues?
related to #3685 

### Screenshots
old table show:
image

new table show:
image

image

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: plugins/starrocks/utils.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ func getTablesByDomainLayer(domainLayer string) []string {
 	switch domainLayer {
 	case "code":
 		return []string{
-			"refs_commits_diffs",
+			"commits_diffs",
```
Comment (warren830) [inline plugins/starrocks/utils.go:29]: now we split refs_commits_diffs to two tables, do we only write down one table here?

File: models/domainlayer/code/commits_diff.go [hunk 2]
```diff
@@ -17,15 +17,22 @@ limitations under the License.
...
+func (CommitsDiff) TableName() string {
+	return "commits_diffs"
+}
+
+type FinishedCommitsDiffs struct {
+	NewCommitSha string `gorm:"primaryKey;type:varchar(40)"`
+	OldCommitSha string `gorm:"primaryKey;type:varchar(40)"`
+}
+
+func (FinishedCommitsDiffs) TableName() string {
```
Comment (warren830) [inline models/domainlayer/code/commits_diff.go:36]: what is this table?

File: models/domainlayer/domaininfo/domaininfo.go [hunk 3]
```diff
@@ -41,7 +41,7 @@ func GetDomainTablesInfo() []Tabler {
 		&code.PullRequestCommit{},
 		&code.PullRequestLabel{},
 		&code.Ref{},
-		&code.RefsCommitsDiff{},
+		&code.CommitsDiff{},
```
Comment (warren830) [inline models/domainlayer/domaininfo/domaininfo.go:44]: now we split refs_commits_diffs to two tables, do we only write down one table here?

File: models/migrationscripts/20221109_modfiy_commits_diffs.go [hunk 4]
```diff
@@ -83,6 +83,7 @@ func (script *modifyCommitsDiffs) Up(basicRes core.BasicRes) errors.Error {
 			dst.CommitSha = s.CommitSha
 			dst.NewCommitSha = s.NewRefCommitSha
 			dst.OldCommitSha = s.OldRefCommitSha
+			dst.SortingIndex = s.SortingIndex
```
Comment (warren830) [inline models/migrationscripts/20221109_modfiy_commits_diffs.go:86]: what is this? can you add some comment here?

File: plugins/refdiff/tasks/commit_diff_calculator.go [hunk 5]
```diff
@@ -34,7 +34,7 @@ func CalculateCommitsDiff(taskCtx core.SubTaskContext) errors.Error {
 	db := taskCtx.GetDal()
 	ctx := taskCtx.GetContext()
 	logger := taskCtx.GetLogger()
-	insertCountLimitOfRefsCommitsDiff := int(65535 / reflect.ValueOf(code.RefsCommitsDiff{}).NumField())
+	insertCountLimitOfCommitsDiff := int(65535 / reflect.ValueOf(code.CommitsDiff{}).NumField())
```
Comment (warren830) [inline plugins/refdiff/tasks/commit_diff_calculator.go:37]: can you write some comment here?

File: plugins/refdiff/tasks/commit_diff_calculator.go [hunk 6]
```diff
@@ -69,7 +69,8 @@ func CalculateCommitsDiff(taskCtx core.SubTaskContext) errors.Error {
 	logger.Info("Create a commit node graph with node count[%d]", commitNodeGraph.Size())
 
 	// calculate diffs for commits pairs and store them into database
-	commitsDiff := &code.RefsCommitsDiff{}
+	commitsDiff := &code.CommitsDiff{}
+	finishedCommitDiff := &code.FinishedCommitsDiffs{}
```
Comment (warren830) [inline plugins/refdiff/tasks/commit_diff_calculator.go:73]: still confusing with this `finishedCommitDiff`

File: plugins/refdiff/tasks/ref_issue_diff_calculator.go [hunk 7]
```diff
@@ -53,23 +53,24 @@ func CalculateIssuesDiff(taskCtx core.SubTaskContext) errors.Error {
...
+		dal.Join("left join refs new_refs on new_refs.commit_sha = commits_diffs.new_commit_sha"),
+		dal.Join("left join refs old_refs on old_refs.commit_sha = commits_diffs.old_commit_sha"),
+		dal.Orderby("new_refs.id ASC"),
+		dal.Where("new_refs.repo_id = ? and pull_request_issues.issue_key > 0 and (new_refs.id, old_refs.id) in ?",
 			repoId, pairList),
-		dal.Select(`refs_commits_diffs.new_ref_commit_sha as new_ref_commit_sha, refs_commits_diffs.old_ref_commit_sha as old_ref_commit_sha, 
+		dal.Select(`commits_diffs.new_commit_sha as new_ref_commit_sha, commits_diffs.old_commit_sha as old_ref_commit_sha, 
 			pull_request_issues.issue_id as issue_id, pull_request_issues.issue_key as issue_number, 
-			refs_commits_diffs.new_ref_id as new_ref_id, refs_commits_diffs.old_ref_id as old_ref_id`),
+			new_refs.id as new_ref_id, old_refs.id as old_ref_id`),
```
Comment (warren830) [inline plugins/refdiff/tasks/ref_issue_diff_calculator.go:73]: I can understand this sql, but this is still too complicated, can you write some comment to explain this?

File: plugins/starrocks/utils.go (hunk 1 above)
Comment (abeizn) [reply plugins/starrocks/utils.go:29]: you are right, I will add another

File: models/domainlayer/code/commits_diff.go (hunk 2 above)
Comment (abeizn) [reply models/domainlayer/code/commits_diff.go:36]: ref_commits_diffs -> commits_diffs and finished_commits_diffs(it records the new_commit_sha and old_commit_sha pair ).

File: models/domainlayer/domaininfo/domaininfo.go (hunk 3 above)
Comment (abeizn) [reply models/domainlayer/domaininfo/domaininfo.go:44]: same to above

File: models/migrationscripts/20221109_modfiy_commits_diffs.go (hunk 4 above)
Comment (abeizn) [reply models/migrationscripts/20221109_modfiy_commits_diffs.go:86]: copy all data, sortingindex is the one.

File: plugins/refdiff/tasks/ref_issue_diff_calculator.go (hunk 7 above)
Comment (abeizn) [reply plugins/refdiff/tasks/ref_issue_diff_calculator.go:73]: ok

File: plugins/refdiff/tasks/commit_diff_calculator.go (hunk 6 above)
Comment (abeizn) [reply plugins/refdiff/tasks/commit_diff_calculator.go:73]: same to above

File: models/domainlayer/code/commits_diff.go [hunk 8]
```diff
@@ -17,15 +17,22 @@ limitations under the License.
...
-	OldRefId        string `gorm:"primaryKey;type:varchar(255)"`
-	CommitSha       string `gorm:"primaryKey;type:varchar(40)"`
-	NewRefCommitSha string `gorm:"type:varchar(40)"`
-	OldRefCommitSha string `gorm:"type:varchar(40)"`
-	SortingIndex    int
+type CommitsDiff struct {
+	CommitSha    string `gorm:"primaryKey;type:varchar(40)"`
+	NewCommitSha string `gorm:"primaryKey;type:varchar(40)"`
+	OldCommitSha string `gorm:"primaryKey;type:varchar(40)"`
+	SortingIndex int
```
Comment (warren830) [inline models/domainlayer/code/commits_diff.go:24]: what is SortingIndex?

File: models/domainlayer/code/commits_diff.go (hunk 2 above)
Comment (warren830) [reply models/domainlayer/code/commits_diff.go:36]: But commit_diff already have those two fields, why do we still need finished_commits_diffs?

Comment (abeizn) [reply models/domainlayer/code/commits_diff.go:36]: The difference between this is that commits_diffs is all the data, and finish_commits_diffs is equivalent to deploying the corresponding sha.

**Review by warren830 — APPROVED** [review]
LGTM

**Pull request merged**